geospaas/utils/processing_base_command.py

```diff
@@ -0,0 +1,153 @@
+                            metavar='YYYY-MM-DD',
+                            default='2100-12-31',
+                            help='End of time range',
+                            type=valid_date)
+
+        parser.add_argument('--extent',
+                            action='store',
+                            metavar=('MIN_LON', 'MAX_LON', 'MIN_LAT', 'MAX_LAT'),
+                            default=None,
+                            help='Spatial extent',
+                            type=float,
+                            nargs=4)
+
+        parser.add_argument('--geojson',
+                            action='store',
+                            metavar='GeoJSON',
+                            default=None,
+                            help='Filename of GeoJSON with polygon for spatial search',
+                            type=str)
+
+        parser.add_argument('--mask',
+                            action='store',
+                            default='',
+                            help='Wildcard for filtering on filenames',
+                            type=str)
+
+        parser.add_argument('--force', action='store_true',
+                            help='Force processing')
+
+        parser.add_argument('--limit',
+                            action='store',
+                            default=None,
+                            type=int,
+                            help='Number of entries to process')
+
+    def geometry_from_options(self, extent=None, geojson=None, **kwargs):
+        """ Generate geometry to use in spatial search
+
+        If extent is not None, return Polygon generated as POLYGON(()) with min/max lon/lat
+        If geojson is not None, return Polygon loaded from the GeoJSON file
+        Otherwise, return None
+
+        Parameters
+        ----------
+        extent : [float, float, float, float]
+            Values of min_lon, max_lon, min_lat, max_lat
+        geojson : str
+            Filename with Polygon GeoJSON
+        **kwargs : dict
+            other options from input arguments
+
+        Returns
+        -------
+        geometry : geos.Polygon or None
+        """
+        polygon = None
+
+        if extent is not None:
+            # generate polygon from input extent parameter
+            polygon = Polygon([ [extent[0], extent[2]],
+                                [extent[1], extent[2]],
+                                [extent[1], extent[3]],
+                                [extent[0], extent[3]],
+                                [extent[0], extent[2]]])
+        elif geojson is not None and os.path.exists(geojson):
+            # try to load GeoJSON and generate valid polygon
+            with open(geojson, 'rt') as f:
+                geojson = f.read().strip()
+            try:
+                polygon = GEOSGeometry(geojson)
+            except (ValueError, GDALException, GEOSException):
+                raise ValueError('Failed to read valid GeoJSON from %s'%options['geojson'])
```

Comment:
It maybe would be better to use CommandError here: https://docs.djangoproject.com/en/2.0/howto/custom-management-commands/ (very bottom of the page)